dns_sd_windows.c

```diff
@@ -47,6 +47,74 @@ static bool is_localhost4(const struct sockaddr_in *saddr)
 		saddr->sin_addr.S_un.S_un_b.s_b4 == 1;
 }
 
+static struct dns_sd_discovery_data *new_discovery_data(struct dns_sd_discovery_data *dd)
+{
+	struct dns_sd_discovery_data *d;
+
+	d = zalloc(sizeof(*d));
+	if (!d)
+		return NULL;
+
+	if (dd)
+		d->lock = dd->lock;
+
+	return d;
+}
+
+static mdns_string_t ipv4_address_to_string(char* buffer, size_t capacity, const struct sockaddr_in* addr,
+	size_t addrlen) {
```

Comment:
Please try to fit everything into 80 columns, as much as possible, attach pointer stars to variables and not types, and align line breaks to the function's first argument with tabs and complete with spaces.

Reply:
Totally understand - was trying to keep most of the things in this file the same as the original file, in this case:
https://github.com/mjansson/mdns/blob/main/mdns.c#L63-L65
to make updates/maintenance less error prone...

I can update - but it will mean that it will be harder to visually inspect things...

Reply:
How much of the library do we share and what prevents us from using mdns.c directly?

Reply:
mdns.c isn't a "library" - it's an example application (which has it's own `main()` function), and declares everything we use as `static`.

We use the header file as is - but since we need our own callback, and need our own transmit we can't use the *.c files from upstream - other than some of the underlying functions (which are the ones you mostly call out at the beginning of the file) - we need to change most of the internal functions - just not those...

Reply:
Updated most things.